### DISCUSSION TITLE
Should Mu show all attached serial devices, not those where we recognize VID/PID?

### DISCUSSION THREAD

**Discussion by dybber** [mu-editor/mu#1320] [Ideas & New Features]
We've had some issues with devices that are hard to recognize, see for instance #1270, which is probably just a bug, but also in other circumstances, we've had users unable to use their device, because we didn't have the VID/PID already included with Mu. In these cases users will have to wait until a new release of Mu before being able to user their devices.

One solution could be to just show all COM-ports/serial-ports in the device-dropdown, while still attempting to select the correct device users will get the option to choose another device and even device which might not be supported by any mode.

**Pros:**
 - Less friction for users of unrecognized devices, as they can start programming their devices right away without filling a bug report

**Cons:**
 - Listing "Unrecognized USB-serial device" in the dropdown menu will be confusing to beginners, and it might even be commonplace to have some kind of COM/serial-device connected, which aren't a micro:bit/ESP/CircuitPython device
 - We will perhaps miss out on reports of new device types, as people will start using Mu for new devices, without sending us a report that the VID/PID should be added

Any thoughts?

**Comment by carlosperate** [comment]
> Listing "Unrecognized USB-serial device" in the dropdown menu will be confusing to beginners, and it might even be commonplace to have some kind of COM/serial-device connected, which aren't a micro:bit/ESP/CircuitPython device

Yeah, I think from my point of view this might be too big of a drawback.
As you mention this can be specially confusing when some PCs/laptops come with devices that show up a serial ports as well. My previous laptop had 2, and then connected my BLE headphones added another one, so I always struggled to find the right port when connecting a serial terminal to a board 😅

If we add the option to add VID/PID values via the settings.json file we can mitigate this a bit, although it's clearly a lot more difficult to figure out the values and add them to a json file than it is to select a different device from a drop down menu.

My other hope is that we can start releasing a lot more frequently, so hopefully waiting for a release might not be that bad.

**Reply by dybber** [reply]
I agree, I think the `settings.json` option is the way to go, and I look forward to being able to do more frequent releases :-) Will be good for users, but also much more motivating for us developers :-)
